ValheimPlus/GameClasses/Player.cs

```diff
@@ -347,12 +349,23 @@ public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructio
                 {
                     bool loads_health = il[i].LoadsField(field_Food_m_health);
                     bool loads_stamina = il[i].LoadsField(field_Food_m_stamina);
+                    bool loads_eitr = il[i].LoadsField(field_Food_m_eitr);
 
-                    if (loads_health || loads_stamina)
+                    if (loads_health || loads_stamina || loads_eitr)
                     {
                         il[i].operand = field_Food_m_item;
                         il.Insert(++i, new CodeInstruction(OpCodes.Ldfld, field_ItemData_m_shared));
-                        il.Insert(++i, new CodeInstruction(OpCodes.Ldfld, loads_health ? field_SharedData_m_food : field_SharedData_m_foodStamina));
+                        if (loads_health)
+                        {
+                            il.Insert(++i, new CodeInstruction(OpCodes.Ldfld, field_SharedData_m_food));
+                        } else if (loads_stamina)
+                        {
+                            il.Insert(++i, new CodeInstruction(OpCodes.Ldfld, field_SharedData_m_foodStamina));
+                        }
+                        else
```

Comment:
Probably put this in an `else if (loads_eitr) ` and have some sort of error handling for everything else (which should be nothing for now, but if at some point something gets added or changed we would see this)

(I stumbled upon this by chance and have not been contributing to Valheim+ yet, so please see my comment with a grain of salt because you know and understand this code much better!)